revocation/ocsp/ocsp.go

```diff
@@ -36,24 +36,18 @@ import (
 	"golang.org/x/crypto/ocsp"
 )
 
-// Purpose is an enum for purpose of the certificate chain whose OCSP status
-// is checked
-type Purpose int
-
-const (
-	// PurposeCodeSigning means the certificate chain is a code signing chain
-	PurposeCodeSigning Purpose = iota
-
-	// PurposeTimestamping means the certificate chain is a timestamping chain
-	PurposeTimestamping
-)
-
 // Options specifies values that are needed to check OCSP revocation
 type Options struct {
-	CertChain        []*x509.Certificate
-	CertChainPurpose Purpose // default value is `PurposeCodeSigning`
-	SigningTime      time.Time
-	HTTPClient       *http.Client
+	CertChain []*x509.Certificate
+
+	// CertChainPurpose is the purpose of the certificate chain. Supported
+	// values are x509.ExtKeyUsageCodeSigning and x509.ExtKeyUsageTimeStamping.
+	// When not provided, the default value x509.ExtKeyUsageAny is also taken as
+	// a code signing certificate chain.
+	CertChainPurpose x509.ExtKeyUsage
```

Comment:
Is the CertChainPurpose field used solely for validating the certificate chain? If so, instead of having a dedicated field for this purpose, could we provide a function that can validate the certificate chain, which the caller will pass? Additionally, could we provide a default implementation of that function, so that callers don't need to worry about the CertChainPurpose field?

Reply:
Tracking in the above comment.